### PR TITLE
Aswathy/DPROD-2811/Make faster builds for covering specific pages locally

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   Make faster builds for covering specific pages locally
- Make the page deploy supported from env
- Page listed in env are deployed seamlessly for localhost
- Add the required router path to the env.development  eg: `GATSBY_BUILD_PAGES='/markets/commodities/'`

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by prince-deriv — DISMISSED** [review]

File: gatsby-node.js [hunk 1]
```diff
@@ -349,41 +349,17 @@ const BuildPage = (page, actions) => {
 exports.onCreatePage = ({ page, actions }) => {
     const { deletePage } = actions
     const isProduction = process.env.GATSBY_ENV === 'production'
-    const pagesToBuild = process.env.GATSBY_BUILD_PAGES || 'all'
+    const pagesToBuild = process.env.GATSBY_BUILD_PAGES
```
Comment (hasan-deriv) [inline gatsby-node.js:352]: is this an array or string?

Comment (prince-deriv) [reply gatsby-node.js:352]: string I guess, but it will be great if this can handle multiple pages like `markets | help-centre`

**Review by mayuran-deriv — APPROVED** [review]

File: gatsby-node.js (hunk 1 above)
Comment (hasan-deriv) [reply gatsby-node.js:352]: Yeah, that's what i am thinking. if it's string then we can't build more than one page.

**Review by konstantinosG-deriv — APPROVED** [review]

File: gatsby-node.js (hunk 1 above)
Comment (aswathy-deriv) [reply gatsby-node.js:352]: Its a string.I don't think so the developer will work on multiple pages at a time.They can access the page by changing the route.But yeah we can add for multiple pages.But it should not increase the memory heap

Comment (dan-deriv) [reply gatsby-node.js:352]: Good callout @hasan-deriv 
It has to allow multiple pages.

Comment (mitra-deriv) [reply gatsby-node.js:352]: I agree to have multiple pages , sometimes one component is shared and can affect on other pages also

Comment (hasan-deriv) [reply gatsby-node.js:352]: Agreed @mitra-deriv

**Review by meenakshi-deriv — DISMISSED** [review]

File: gatsby-node.js [hunk 2]
```diff
@@ -349,48 +349,40 @@ const BuildPage = (page, actions) => {
...
-            'careers',
-            // 'markets',
-            // 'trade-types' Note: Feel free to adjust pages you want to skip building for faster local development
-        ],
-    }
-
-    const disallowedPages = pagesCategory[pagesToBuild] || []
-
-    const regex = new RegExp(`/${disallowedPages.join('|') + '|'}/g`)
+        console.log('pages', pages)
```
Comment (mitra-deriv) [inline gatsby-node.js:368]: why do we need this console log?  🙏

Comment (aswathy-deriv) [reply gatsby-node.js:368]: We need to keep that for now.To see the page in the terminal console.

**Review by mayuran-deriv — APPROVED** [review]

**Review by mitra-deriv — APPROVED** [review]

**Review by hasan-deriv — APPROVED** [review]

File: gatsby-node.js [hunk 3]
```diff
@@ -349,48 +349,40 @@ const BuildPage = (page, actions) => {
...
-    const pagesToBuild = process.env.GATSBY_BUILD_PAGES || 'all'
+    const pagesToBuild = process.env.GATSBY_BUILD_PAGES
+    if (pagesToBuild) {
+        const pages_loaded = pagesToBuild.split(',')
+        const allowed_pages = ['', pages_loaded]
 
-    // First delete the incoming page that was automatically created by Gatsby
-    // So everything in src/pages/
-    deletePage(page)
+        const pages = allowed_pages.reduce((result, Item) => {
```
Comment (mayuran-deriv) [inline gatsby-node.js:357]: good usage of reduce :fire:

Comment (aswathy-deriv) [reply gatsby-node.js:357]: thanks @mayuran-deriv

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**